.github/workflows/check-fuzz.yml

```diff
@@ -0,0 +1,25 @@
+# Smoke test to build fuzz targets.
+# Deserves its own action given that it depends on nightly
+# and there's currently no way to define multiple toolchains through the 
+# `rust-toolchain.toml` configuration file.
+name: Build Fuzz Targets
+on:
+  push:
+    branches:
+      - main
+    pull_request:
+
+jobs:
+  check-fuzz:
+    name: Build fuzz targets
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install nightly
+
+        run: rustup install nightly-2024-07-07
```

Comment:
Pinning to the most recent nightly version, and the one that I used locally for testing.